src/pages/home/HeaderView.js

```diff
@@ -239,7 +239,7 @@ export default compose(
             canEvict: false,
         },
         parentReport: {
-            key: ({report}) => `${ONYXKEYS.COLLECTION.REPORT}${report.parentReportID}`,
+            key: ({report}) => `${ONYXKEYS.COLLECTION.REPORT}${report.parentReportID || report.reportID}`,
```

Comment:
I don't know if this change makes sense - the parentReport is never the report.reportID right? Like can we just return nothing here if it doesn't exist?

Reply:
Was wondering the same, we should not retrun a wrong report as parent. I think we should return 0 in this case and make sure we check for that -> its root does not have a parent

Reply:
Without this, Onyx will return all the reports in the report collection  if `report.parentReportID` is undefined because the key will be `reports_`. I spoke to @tgolen about this and we agreed that falling back to the report was a good solution in this case. As I mentioned in the OP, we could probably consolidate these into a single Onyx key `report` so the component always gets the right data and doesn't need to have any logic to decide which report it should use.

Reply:
> Without this, Onyx will return all the reports in the report collection if report.parentReportID is undefined because the key will be report_. 

Does that also happen with report_0?

Reply:
Not sure, but lately I've started seeing a `report_0` key in Onyx (which seems to be coming from PHP somewhere) so I assume that'd be returned instead.

![Screenshot 2023-05-17 at 11 58 43 AM](https://github.com/Expensify/App/assets/22219519/b3ed34b4-1b3f-4e1e-abe5-de30b1d527af)

Reply:
> Does that also happen with report_0?

I don't think it should, no. It's expected that if you subscribe to a key like `report_` that it gives you the entire collection and that is normal behavior. The `undefined` reportID is a nasty gotcha that results in subscribing to the entire collection.